### PR TITLE
Update branch policy

### PR DESCRIPTION
**Pull request by sneridagh**

### CODE REVIEW

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [inline docs/source/contributing/branch-policy.md:40]: I'd simplify a bit. I'm also concerned about the loose use of "legacy" throughout this policy. I think we need to explicitly define "legacy" as a term for Volto and all of Plone.

```
:   `17.x.x` is the current stable and actively developed branch in git.
    It may receive new features and bug fixes.
    Upon the final release of version `18.0.0`, the `17.x.x` branch line will become legacy.
    Any new feature will be generally released to the `main` branch, and only backported to older versions if the Volto Team approves it.
```

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [inline docs/source/contributing/branch-policy.md:43]: See https://github.com/plone/volto/pull/4905 and https://github.com/plone/volto/issues/4904

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [inline docs/source/contributing/branch-policy.md:45]:
```suggestion
    Any new feature will be released generally to the `main` branch, and only backported to `16.x.x` if the Volto Team approves it.
```

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [inline docs/source/contributing/branch-policy.md:48]:
```suggestion
:   `15.x.x` and older branches are considered legacy branches and won't receive any updates.
```

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [inline docs/source/contributing/branch-policy.md:49]:
```suggestion
    If you need a bug fix or security update to any of them, please submit a pull request, and the Volto Team will review it to determine whether it's suitable to merge.
```

File: news/5363.documentation
Comment (stevepiercy) [inline news/5363.documentation:1]:
```suggestion
Updated branch policy. @sneridagh
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
"legacy" is used loosely throughout. Does it mean "won't receive any update" or "will receive bugfixes"?

File: docs/source/contributing/branch-policy.md
Comment (sneridagh) [reply docs/source/contributing/branch-policy.md:40]: It's the term we decided during the Volto Team Meeting at the conf. Don't know if we have to rethink it, but let's discuss it. I think @davisagli gave his ok during the meeting.
No, we decided that the "legacy" versions won't receive anything, unless it's justified and approved by the Team. 
This is to reduce the overall burden on the Release Manager (eg. when we have five legacy versions).

File: docs/source/contributing/branch-policy.md
Comment (sneridagh) [reply docs/source/contributing/branch-policy.md:43]: We also talked about this one... although is controversial, we decided to stick with the denomination not as a LTS per se, but to align with the Plone LTS (6.0).

File: docs/source/contributing/branch-policy.md
Comment (sneridagh) [reply docs/source/contributing/branch-policy.md:40]: jeez I just remembered you couldn't attend :( It would have been great to have you there, although few things, they were important. It's only now that I had time to write them down (and I'm cutting the 17.x.x branch now too).

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [reply docs/source/contributing/branch-policy.md:43]: I think it would be clearer to say this, and avoid "LTS".

```suggestion
:   This version of Volto has the [same maintenance and support schedule as Plone 6.0](https://plone.org/download/release-schedule).
```

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [reply docs/source/contributing/branch-policy.md:40]: Why do we need YAOT (yet another obfuscating term)?

Let's just call it "unsupported" or "no longer supported" (which would be consistent with the term already used in the [Plone Release Schedule](https://plone.org/download/release-schedule)), and drop "legacy".

Comment (sneridagh) [reply docs/source/contributing/branch-policy.md:40]: +1 good enough for me

/cc @plone/volto-team @davisagli thoughts? If any, we will go with it.

Comment (stevepiercy) [reply docs/source/contributing/branch-policy.md:40]: Updated suggestion, replacing "legacy" with "no longer supported".

File: docs/source/contributing/branch-policy.md
Comment (stevepiercy) [inline docs/source/contributing/branch-policy.md:48]:
```suggestion
:   `15.x.x` and older branches are no longer supported.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Replace "legacy" with "no longer supported".

File: docs/source/contributing/branch-policy.md
Comment (sneridagh) [reply docs/source/contributing/branch-policy.md]: Done.

**Comment by sneridagh** [comment]
@stevepiercy give me your final ack to merge.

File: packages/registry/CHANGELOG.md
Comment (stevepiercy) [inline packages/registry/CHANGELOG.md:6]: This has moved.

```suggestion
     https://6.docs.plone.org/contributing/index.html#contributing-change-log-label
```

File: packages/registry/CHANGELOG.md
Comment (stevepiercy) [inline packages/registry/CHANGELOG.md:3]:
```suggestion
<!-- Do *NOT* add new change log entries to this file.
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
New language looks good.

However I found a couple of things that needed fixing after you pushed commits from other branches. Once those are fixed, then I approve. Thank you!

**Comment by sneridagh** [comment]
@stevepiercy uh, why that changes appear in here? What the heck happened?

**Review by stevepiercy — APPROVED** [review]
Indeed, git gremlins. LGTM. Thank you!

**Comment by sneridagh** [comment]
@stevepiercy thanks to you!!

File: docs/source/contributing/branch-policy.md
Comment (davisagli) [inline docs/source/contributing/branch-policy.md:39]: @sneridagh The word "released" is confusing here. You mean "merged", right?

Comment (stevepiercy) [reply docs/source/contributing/branch-policy.md:39]: How about this?

```
    Any new feature will be merged into the `main` branch, and only backported to and released in older versions, if the Volto Team approves it.
```

**Pull request merged**
